hasher-matcher-actioner/hmalib/common/models/bank.py

```diff
@@ -378,3 +539,59 @@ def remove_bank_member(self, bank_id: str, bank_member_id: str):
         )
         bank_member.is_removed = False
         bank_member.write_to_table(self._table)
+
+    def add_bank_member_signal(
+        self,
+        bank_id: str,
+        bank_member_id: str,
+        signal_type: t.Type[SignalType],
+        signal_value: str,
+    ) -> BankMemberSignal:
+        """
+        Adds a BankMemberSignal entry. First, identifies if a signal for the
+        corresponding (type, value) tuple exists, if so, reuses it, it not,
+        creates a new one.
+
+        Returns a BankMemberSignal object. Clients **should not** care whether
+        this is a new signal_id or not.
+
+        This check is being done here because signal uniqueness is enforced by
+        the same table. If this were being done in a different table/store, we
+        could be doing the check at a different layer eg.
+        hmalib.banks.bank_operations.
+        """
+        # First, we get a unique signal_id!
+        signal_id = BankedSignalEntry.get_unique(
+            self._table, signal_type=signal_type, signal_value=signal_value
+        ).signal_id
+
+        # Next, we create the bank member signal
+        member_signal = BankMemberSignal(
+            bank_id=bank_id,
+            bank_member_id=bank_member_id,
+            signal_id=signal_id,
+            signal_type=signal_type,
+            signal_value=signal_value,
+            updated_at=datetime.now(),
+            status=BankMemberSignalStatus.PENDING,
+        )
+        member_signal.write_to_table(self._table)
+        return member_signal
+
+    def unmark_bank_member_signal(self, bank_id: str, signal_id: str):
+        """
+        Remove the bank-member-signal from the pending index.
+        """
+        BankMemberSignal.unmark(self._table, bank_id=bank_id, signal_id=signal_id)
+
+    def get_bank_member_signals_to_process(
```

Comment:
nit: docstring or at least a comment for the reader how this might not be full set and value wont change unless some are 'processed' ?